src/os/KeyValueStore.cc

```diff
@@ -1026,6 +1020,8 @@ int KeyValueStore::queue_transactions(Sequencer *posr, list<Transaction*> &tls,
     dump_transactions(o->tls, o->op, osr);
   dout(5) << "queue_transactions (trailing journal) " << " " << tls <<dendl;
   queue_op(osr, o);
+  do_op(osr, *handle);
+  finish_op(osr);
```

Comment:
I have a similar patch, which eliminates the osr also. it completes the request in same thread context(sharded worker thread),